src/iris/role_lookup/oncall.py

```diff
@@ -28,7 +28,8 @@ def call_oncall(self, url, expected_type):
         try:
             r = self.requests.get(url)
         except Exception:
-            stats['oncall_error'] += 1
```

Comment:
Could we avoid these checks by making `stats` a colletions.defaultdict(int) or a collections.Counter()?

Reply:
@jrgp we could, but is there an advantage? As it stands right now when api.py uses oncall.py through role_lookup the stats dictionary will be empty and never used for anything since api.py is not set up to emit metrics (unlike sender.py). Is there any use to make it so we can increment a field in a dictionary that is never going to be used? My thought was that this way it is clear that this is a metric that is not always used at the moment. That being said I wouldn't particularly mind doing it either way.